src/libutil/ustring.cpp

```diff
@@ -501,8 +565,11 @@ ustring::getstats(bool verbose)
 size_t
 ustring::hash_collisions(std::vector<ustring>* collisions)
 {
-    UstringTable& table(ustring_table());
-    return table.find_hash_collisions(collisions);
+    std::lock_guard<std::mutex> lock(collision_mutex);
```

Comment:
Leave the previous implementation in place for when `PREVENT_HASH_COLLISIONS == 0` ? (and ifdef out the reverse map definition, etc...) ?

Reply:
When PREVENT_HASH_COLLISIONS == 0, the reverse map, etc. is not used. (And also you can't ask if there were collisions. Is that what you wanted?)